src/HDF5.jl

```diff
@@ -302,7 +302,21 @@ struct Opaque
 end
 
 # An empty array type
-struct EmptyArray{T} end
+struct EmptyArray{T} <: AbstractArray{T,0} end
+# Required AbstractArray interface
+Base.size(::EmptyArray) = ()
+Base.IndexStyle(::EmptyArray) = IndexLinear()
+Base.getindex(::EmptyArray, ::Int) = error("cannot index an `EmptyArray`")
+Base.setindex!(::EmptyArray, v, ::Int) = error("cannot assign to an `EmptyArray`")
+# Optional interface
+Base.similar(::EmptyArray{T}) where {T} = EmptyArray{T}()
```

Comment:
Any specific reason to include these optional interface methods? I guess it doesn't hurt in case someone want's to use them, albeit the `EmptyArray` type is quite limited in utility.

Reply:
I included them because some recent work I've been doing quite often relies on getting `similar` arrays, so it was on my mind as something useful. For instance, I could imagine making a type-compatible working buffer based on a loaded dataset, even if the "loaded" data is the empty set.